Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+
+        public bool InverseMatching { get; }
+        public IList<string> CharacterSpecifiers { get; }
+        private readonly string specifier;
+
+        public CharacterClass(string specifier)
+        {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a character class");
+            }
+            this.specifier = specifier;
+            string actualSpecifier = m.Groups["expression"].Value;
+            InverseMatching = actualSpecifier.StartsWith("^");
+            CharacterSpecifiers = new List<string>();
+
+            ExtractCharacterSpecifiers(InverseMatching ? actualSpecifier.Substring(1) : actualSpecifier);
+        }
+
+        public string Specifier
+        {
+            get
+            {
+                return specifier;
+            }
+        }
+
+        private static readonly Regex CharacterRanges = new Regex(@"(\\[dDwWsS]|(\\[ntfvr]|\\([0-7]{3}|x[\dA-F]{2}|u[\dA-F]{4}|[\\\.\[\]])|.)(-(\\[ntfvr]|\\([0-7]{3}|x[A-F]{2}|u[\dA-F]{4}|[\.\\\[\]])|.))?)");
+        private void ExtractCharacterSpecifiers(string characterClass)
+        {
+            MatchCollection specifiers = CharacterRanges.Matches(characterClass);
+            
+            foreach (Match specifier in specifiers)
+            {
+                if (specifier.Value.Contains("\\"))
+                {
+                    if (specifier.Value.EndsWith("-\\"))
+                    {
+                        // BOOM!
+                        throw new ArgumentException("Character Ranges that have incorrectly escaped characters as target are not allowed");
+                    }
+                    else if (specifier.Value.Length == 1)
+                    {
+                        // fun... we somehow got to grab a single backslash. Pattern is probably broken
+                        // alas for simplicity we just skip the incorrect backslash
+                        // TODO: make a warning from this.. how?? no idea
+                        continue;
+                    }
+                }
+                CharacterSpecifiers.Add(specifier.Value);
+            }
+        }
+
+        public string Description
+        {
+            get
+            {
+                return string.Format(InverseMatching 
+                    ? AssistantResources.AtomDescription_CharacterClass_Inverted 
+                    : AssistantResources.AtomDescription_CharacterClass
+                    , HumanReadableClass());
+            }
+        }
+
+        private string HumanReadableClass()
+        {
+            return string.Join(", ", CharacterSpecifiers); // join last with and?
+        }
+
+        public bool TryMatch(ref string text)
+        {
+            throw new NotImplementedException();
```

Comment:
Those methods are for actually running the IRegularExpression against a given string. I just didn't do that yet. 
The design for that is up for debate though since I want to write a "debugger"-like functionality in [future-version] the `TryMatch`could be removed for now, but I want something to remind me of this when I work on the thing